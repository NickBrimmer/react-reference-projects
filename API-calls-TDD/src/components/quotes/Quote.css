.Quote {
  max-width: 200px;
  overflow: hidden;
  text-align: center;
  img {
    max-height: 200px;
  }
}
