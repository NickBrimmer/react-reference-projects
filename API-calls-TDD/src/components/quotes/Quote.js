import React from 'react';
import PropTypes from 'prop-types';
import styles from './Quote.css';

const Quote = ({ name, text, image }) => {
  return (
    <figure className={styles.Quote}>
      <img src={image} alt={name} />
      <figcaption>{name}</figcaption>
      <p>{text}</p>
    </figure>
  );
};

Quote.propTypes = {
  name: PropTypes.string.isRequired,
  text: PropTypes.string.isRequired,
  image: PropTypes.string.isRequired
};

export default Quote;
