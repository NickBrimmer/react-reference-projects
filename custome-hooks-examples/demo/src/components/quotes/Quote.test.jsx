import React from 'react';
import { shallow } from 'enzyme';
import Quote from './Quote';

describe('Quote component', () => {
  it('renders Quote', () => {
    const wrapper = shallow(<Quote character="Bender" text="hello" image="https://image.com" />);
    expect(wrapper).toMatchSnapshot();
  });
});
