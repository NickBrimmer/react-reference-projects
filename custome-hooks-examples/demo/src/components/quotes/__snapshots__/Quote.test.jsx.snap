// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quote component renders Quote 1`] = `
<section
  className="Quote"
>
  <img
    src="https://image.com"
  />
  <p>
    Bender
  </p>
  <p>
    "
    hello
    "
  </p>
</section>
`;
